web/src/beta/components/Modal/index.tsx

```diff
@@ -1,54 +1,163 @@
-import { styled, useTheme } from "@reearth/services/theme";
+import { styled } from "@reearth/services/theme";
+
+type Size = "sm" | "md" | "lg";
+
+export type SidebarTab = {
+  id: string;
+  label: string;
+  content: ReactNode;
+  tabButton1?: ReactNode;
+  tabButton2?: ReactNode;
+};
 
 type Props = {
   className?: string;
-  title?: string;
-  size?: "sm" | "md" | "lg";
+  modalTitle?: string;
+  size?: Size;
   button1?: ReactNode;
   button2?: ReactNode;
   children?: ReactNode;
   isVisible?: boolean;
   onClose?: () => void;
+  showSidebar?: boolean;
+  sidebarTabs?: SidebarTab[];
 };
 
 const Modal: React.FC<Props> = ({
   className,
-  title,
-  size,
+  modalTitle,
+  size = "md",
   button1,
   button2,
+  children,
   isVisible,
   onClose,
-  children,
+  sidebarTabs = [],
 }) => {
-  const theme = useTheme();
+  const [selectedTab, setSelectedTab] = useState<string | null>(
+    sidebarTabs.length > 0 ? sidebarTabs[0].id : null,
+  );
+
+  const handleTabChange = (tabId: string) => {
+    setSelectedTab(tabId);
+  };
+
   return (
-    <Wrapper className={className} size={size} isVisible={isVisible} onClose={onClose}>
-      <Title size="h4" weight="bold" color={theme.general.content.strong}>
-        {title}
-      </Title>
-      {children}
-      <ButtonWrapper>
-        {button2}
-        {button1}
-      </ButtonWrapper>
+    <Wrapper
+      className={className}
+      size={size}
+      isVisible={isVisible}
+      modalTitle={modalTitle}
+      onClose={onClose}>
+      {sidebarTabs.length > 0 ? (
+        <SidebarWrapper>
+          <NavBarWrapper>
+            {sidebarTabs.map(tab => (
+              <Tab
+                key={tab.id}
+                isSelected={selectedTab === tab.id}
+                onClick={() => handleTabChange(tab.id)}>
+                {tab.label}
+              </Tab>
+            ))}
+          </NavBarWrapper>
+          <SidebarContentWrapper>
+            <TabContent>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.content}
+            </TabContent>
+            <TabButtonWrapper>
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton1}
+              {selectedTab && sidebarTabs.find(tab => tab.id === selectedTab)?.tabButton2}
+            </TabButtonWrapper>
+          </SidebarContentWrapper>
+        </SidebarWrapper>
+      ) : (
+        <>
+          <ContentWrapper>{children}</ContentWrapper>
+          <ButtonWrapper>
+            {button2}
+            {button1}
+          </ButtonWrapper>
+        </>
+      )}
     </Wrapper>
   );
 };
 
-const Title = styled(Text)`
-  text-align: center;
-  margin-bottom: 24px;
+const SidebarWrapper = styled.div`
+  display: flex;
+  align-items: flex-start;
+  align-self: stretch;
+`;
+
+const NavBarWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 16px;
+  align-items: flex-start;
+  gap: 10px;
+  align-self: stretch;
```

Comment:
Are you trying to make sure the element takes up the full height/width? If so, flex: 1 should be good. I think stretch might have unforeseen results depending on the content

Reply:
The primary intention of the NavBarWrapper is to take up the full height of its parent container and expand vertically to accommodate the tabs. However, using flex: 1 would make the NavBarWrapper expand both horizontally and vertically, which is not desired. It would cause the NavBarWrapper to take up more width than needed, potentially affecting the layout and alignment of other elements within its parent container. 
![image](https://github.com/reearth/reearth/assets/89770889/c9b9f802-dacb-4730-8c3e-2568d6444162)

Reply:
done